Cocoaheads School_Lesson2.playground/Contents.swift

```diff
@@ -19,31 +19,60 @@ class Stack {
 }
 
 class StackStatistics: Stack {
-    private var minValues = [Int]()
-
+    private var minElement: Int?
+    
     override func push(element: Int) {
-        super.push(element: element)
-        if let minValue = minValues.last {
+        if let minValue = minElement {
             if minValue > element {
-                minValues.append(element)
+                super.push(element: (2*element - minValue))
+                minElement = element
             } else {
-                minValues.append(minValue)
+                super.push(element: element)
             }
         } else {
-            minValues.append(element)
+            minElement = element
+            super.push(element: element)
         }
     }
     
     override func pop() -> Int? {
-        guard !minValues.isEmpty else {
-            return nil
+        if let elementTop = super.top() {
+            if let minValue = minElement {
+                if elementTop > minValue {
+                    super.pop()
+                    return elementTop
+                } else {
+                    let elementValue = minElement
+                    minElement = 2*(minValue) - elementTop
```

Comment:
Why do you use parentheses?

I'd prefer to see:

```suggestion
                    minElement = (2 * minValue) - elementTop
```
or

```suggestion
                    minElement = 2 * minValue - elementTop
```

Reply:
done